### DISCUSSION TITLE
Branch Protection Rules on Private Repo

### DISCUSSION THREAD

**Discussion by natemacfadden** [community/community#53683] [Code Security]
To use branch protection rules on a private repo, it seems like one must upgrade to Github teams (or higher). Must all collaborators have a Github teams account? Or just the owner of the repo?

**Comment by patrick-knight** [comment]
The team will need to create an org, license the correct number of contributors and invite them to the repo. So the individual accounts don't need a teams plan, just the team is responsible for that. You can read more [here](https://docs.github.com/en/get-started/onboarding/getting-started-with-github-team).

**Comment by kavix** [comment]
Hi @natemacfadden,

No — collaborators do not need GitHub Teams accounts.

Branch protection rules are enforced at the repository level, not at the individual user subscription level.

Key point:

Only the repo owner (or organization) needs to be on a plan that supports branch protection for private repositories.

How it works in practice:

* Repo owner/org: must have a plan that includes branch protection features (e.g., Teams/Enterprise depending on setup)
* Collaborators: can use free GitHub accounts
* External contributors: also fine, as long as they have repo access

What collaborators actually see:

They simply follow the rules enforced on the branch, such as:

* Required pull request reviews
* No direct pushes to protected branches
* Required status checks (CI)

They don’t need any paid subscription to participate.

Simple way to think about it:

Branch protection is like security rules on a building.
Only the building owner pays for the security system — visitors just follow the rules.

**Reply by kavix** [reply]
If this answer helped resolve your issue, please consider marking the discussion as answered. It helps others with similar problems find the solution more easily. 🙂
